### PR TITLE
Active directory SSO

### PR DESCRIPTION
**Pull request by Stephen-ONeil**
As implemented in ITAP.

### CODE REVIEW

**Comment by Stephen-ONeil** [comment]
Will have to "test in prod" on this one. The SSO endpoint can't be tested locally as it checks for the prod host name, and I can't deploy this before merging as flux's auto-reconciliation will force the cluster state back to whatever's on the main branch.

File: server/cpho/jinja2/login.jinja2
Comment (Stephen-ONeil) [inline server/cpho/jinja2/login.jinja2:26]: Could delete this, and the corresponding translation entry, if you don't expect to need a user agreement on this project. Don't know if ITAP has it because the client requested it, or if it was required during approvals (and therefore is needed here too).

Comment (AlexCLeduc) [reply server/cpho/jinja2/login.jinja2:26]: Go ahead and delete it, let's try and slip by without it, ITAP was special

File: server/server/middleware.py
Comment (AlexCLeduc) [inline server/server/middleware.py:17]: do we also need `phac_aspc_helper_authorize` ? I'm honestly not sure but we whitelisted both in cedars

**Review by AlexCLeduc — APPROVED** [review]
Looks great, thanks!

File: server/server/middleware.py
Comment (Stephen-ONeil) [reply server/server/middleware.py:17]: Ah, yup, looks like it would be necessary, good catch!

**Pull request merged**
